healthcare/v1/src/test/java/snippets/healthcare/FhirResourceTests.java

```diff
@@ -0,0 +1,261 @@
+import snippets.healthcare.fhir.resources.FhirResourceDeletePurge;
+import snippets.healthcare.fhir.resources.FhirResourceGet;
+import snippets.healthcare.fhir.resources.FhirResourceGetHistory;
+import snippets.healthcare.fhir.resources.FhirResourceGetMetadata;
+import snippets.healthcare.fhir.resources.FhirResourceGetPatientEverything;
+import snippets.healthcare.fhir.resources.FhirResourceListHistory;
+import snippets.healthcare.fhir.resources.FhirResourcePatch;
+import snippets.healthcare.fhir.resources.FhirResourceSearchGet;
+import snippets.healthcare.fhir.resources.FhirResourceSearchPost;
+
+@RunWith(JUnit4.class)
+public class FhirResourceTests {
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String REGION_ID = "us-central1";
+  private static final Gson gson = new Gson();
+
+  private static String fhirStoreName;
+  private static String datasetName;
+
+  private String fhirResourceId;
+  private String fhirResourceName;
+
+  private static String patientType = "Patient";
+
+  private final PrintStream originalOut = System.out;
+  private ByteArrayOutputStream bout;
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(
+        System.getenv(varName),
+        String.format("Environment variable \"%s\" is required to perform these tests.", varName));
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    String datasetId = "dataset-" + UUID.randomUUID().toString().replaceAll("-", "_");
+    datasetName = String.format(
+        "projects/%s/locations/%s/datasets/%s",
+        PROJECT_ID,
+        REGION_ID,
+        datasetId);
+    DatasetCreate.datasetCreate(PROJECT_ID, REGION_ID, datasetId);
+
+    String fhirStoreId = "fhir-" + UUID.randomUUID().toString().replaceAll("-", "_");
+    fhirStoreName = String.format("%s/fhirStores/%s", datasetName, fhirStoreId);
+    FhirStoreCreate.fhirStoreCreate(datasetName, fhirStoreId);
+  }
+
+  @AfterClass
+  public static void deleteTempItems() throws IOException {
+    DatasetDelete.datasetDelete(datasetName);
+  }
+
+  @Before
+  public void beforeTest() throws IOException, URISyntaxException {
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    FhirResourceCreate.fhirResourceCreate(fhirStoreName, patientType);
+
+    Matcher idMatcher = Pattern.compile("\"id\": \"([^\"]*)\",").matcher(bout.toString());
+    if (idMatcher.find()) {
+      fhirResourceId = idMatcher.group(1);
+      fhirResourceName = String.format("%s/fhir/%s/%s", fhirStoreName, patientType, fhirResourceId);
+    }
+
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(originalOut);
+    bout.reset();
+  }
```

Comment:
Ok for now, but at some point it would be good if you delete all the resources you create via these tests.